projects/dummy_PETR/mmdet3d_plugin/structures/utils.py

```diff
@@ -0,0 +1,399 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from logging import warning
+
+import numpy as np
+import torch
+
+import mmdet3d
+from mmdet3d.utils import array_converter
+
+
+@array_converter(apply_to=('val', ))
+def limit_period(val, offset=0.5, period=np.pi):
+    """Limit the value into a period for periodic function.
+
+    Args:
+        val (torch.Tensor | np.ndarray): The value to be converted.
+        offset (float, optional): Offset to set the value range.
```

Comment:
This file is repeated with `mmdet3d/structures/bbox_3d/utils.py`

Reply:
I just temporary put some bbox encoder and decoder code here, and I'm considering move those bbox encoder and decoder under coders/